core/string/char_utils.h

```diff
@@ -31,7 +31,25 @@
 #ifndef CHAR_UTILS_H
 #define CHAR_UTILS_H
 
-#include "core/typedefs.h"
+#include "char_range.h"
```

Comment:
Not too important but we usually name such data-only includes with `.inc` extension.